assets/js/app.js

```diff
@@ -1,122 +1,147 @@
-	for(var key in characters) {
-		console.log(characters[key]);
-		createCharacterCard(characters[key], "#characterArea");
+/**
+ * [description]
+ * @return {[type]} [description]
+ */
+const character = (function(){
+	/**
+	 * [createCharacter description]
+	 * @return {[type]} [description]
+	 */
+	function createCharacter() {
+		for(var key in characters) {
+			console.log(characters[key]);
+			createCharacterCard(characters[key], "#characterArea");
+		}
+	};
+	/**
+	 * [createCharacterCard description]
+	 * @param  {String} character [description]
+	 * @param  {String} cardArea  [description]
+	 * @return {[type]}           [description]
+	 */
+	function createCharacterCard(character, cardArea){
+		// Create a div for each character's card containing name, healthPoints, and image.
+		const $characterDiv = $("<div>", {"class": "characters"})
+		.attr("data-name", character.name);
+		const $characterName = $("<p>")
+		.attr("class", "characterName")
+		.text(character.name);
+		const $characterImage = $("<img>")
+		.attr("class", "characterImage")
+		.attr("src", character.image);
+		const $characterHealthPoints = $("<p>")
+		.attr("class", "characterHealthPoints")
+		.text(character.healthPoints);
+
+		$characterDiv
+		.append($characterName)
+		.append($characterHealthPoints)
+		.append($characterImage);
+
+		$(cardArea).append($characterDiv);
+	};
+
+	/**
+	 * [setCharacterSelection description]
+	 * @param {[type]} enemyArray [description]
+	 */
+	function setCharacterSelection (enemyArray) {
+		console.log("setCharacterSelection function!");
+		enemyArray.forEach(function(enemyCard) {
+			console.log("move the enemies!");
+			console.log(enemyCard);
+			createCharacterCard(enemyCard, "#enemyArea");
+			console.log("enemies moved!");
+			$("#header-text").text("Your Character");
+		})
+	};
+	/**
+	 * [init description]
+	 * @return {[type]} [description]
+	 */
+	function init() {
+		createCharacter();
+	};
+
+	return {
+		createCharacter,
+		createCharacterCard,
+		defender: '',
+		enemies : [],
+		init,
+		setCharacterSelection,
+		yourCharacter: ''
 	}
-}
-
-createCharacter();
-
-// Now I need to declare my variables and create an onclick event for the character cards so that the selected character remains and the rest gets pushed into an array and appends to the 'enemies to attack' div. 
-let yourCharacter;
-const enemies = [];
-let defender; // this is the opponent
-// const healthPoints;
-// const enemyAttackBackPoints;
-
-
-function setCharacterSelection (enemyArray) {
-	console.log("setCharacterSelection function!");
-	enemyArray.forEach(function(enemyCard) {
-		console.log("move the enemies!");
-		console.log(enemyCard);
-		createCharacterCard(enemyCard, "#enemyArea");
-		console.log("enemies moved!");
-		$("#header-text").text("Your Character");		
-	})
-};
 
+}())
+
+
+// Executes code when DOM has fully loaded
+$(document).ready(function() {
+	character.init()
+
+// Create the character object to hold the characters data
 
```

Comment:
I would personally have a initializeEventListeners or addEventListeners function where I would put all the event listeners in rather than just stick them in the document.ready function

Reply:
Thanks for the pro tip! This is definitely something I will begin to implement within my apps. Especially on the new job.